CIP-onchain-cert-metadata/README.md

```diff
@@ -0,0 +1,489 @@
+    "scripts"
+  ]
+}
+```
+
+### Example
+
+```json
+{
+  "subject": "d684512ccb313191dd08563fd8d737312f7f104a70d9c72018f6b0621ea738c5b8213c8365b980f2d8c48d5fbb2ec3ce642725a20351dbff9861ce9695ac5db8",
+  "schemaVersion": "1.0.0",
+  "certificationLevel": 1,
+  "certificateIssuer": {
+    "name": "Audit House LLC",
+    "logo": "https://www.example.com/media/logo.svg"
+    "social": {
+        "contact": "contact@example.com",
+        "link": "https://example.com",
+        "twitter": "twiterHandle",
+        "github": "githubHandle",
+        "website": "https://www.example.com"
+    }
+  },
+  "report": {
+    "reportURLs": [
+        "https://example.io/certificate.pdf",
+        "ipfs://bafybeiemxfal3jsgpdr4cjr3oz3evfyavhwq"
+        ],
+    "reportHash": "c6bb42780a9c57a54220c856c1e947539bd15eeddfcbf3c0ddd6230e53db5fdd"
+  },
+  "summary": "This is the summary of the report."
+  "disclaimer": "This is the legal disclaimer from the report",
+  "scripts": [
+        {
+          "smartContractInfo": {
+            "era": "basho",
+            "compiler": "plutusTx",
+            "compilerVersion": "1.3.0",
+            "optimizer": "plutonomy",
+            "optimizerVersion": "v0.20220721",
+            "progLang": "plutus-v2",
+            "repository": "https://github.com/DAppDev/TestDApp/"
+          },
+          "fullScriptHash": "711dcb4e80d7cd0ed384da5375661cb1e074e7feebd73eea236cd68192",
+          "scriptHash": "1dcb4e80d7cd0ed384da5375661cb1e074e7feebd73eea236cd68192",
+          "contractAddress": "addr1wywukn5q6lxsa5uymffh2esuk8s8fel7a0tna63rdntgrysv0f3ms"
+        },
+        {
+          "smartContractInfo": {
+            "era": "basho",
+            "compiler": "plutusTx",
+            "compilerVersion": "1.3.0",
+            "optimizer": "plutonomy",
+            "optimizerVersion": "v0.20220721",
+            "progLang": "plutus-v2",
+            "repository": "https://github.com/DAppDev/TestDApp/"
+          },
+          "fullScriptHash": "384da5375661cb1e07713eea236cd681921dcb4e80d7cd0ed4e7feebd7",
+          "scriptHash": "6cd68191dcb4e80d7c5661cb1e074e7feebd73eea232d0ed384da537",
+          "contractAddress": "addr1wywukn5q6lrdntgrysv0f7a0tna63ymffh23ms5uxsaesuk8s8fel"
+        }
+  ]
+}
+```
+
+The metadata should be discoverable by all certification stakeholders, including end-users, DApp developers, and ecosystem components, such as light wallets and DApp stores. Information should be indexable by certification issuer, DApp developer, DApp and DApp version.
+
+It should be possible for there to be multiple versions of metadata published by the same certification issuer for the same version of a DApp, particularly when a (minor) update of the evidence is necessary. In such a case it should be possible to identify the most recent version of the report for that DApp version. It will also be the case that the same DApp may have certification metadata provided by multiple different certification issuers.
```

Comment:
In CIP-72, you can register a new update and provide all the script hashes and addresses. If this registration info is up to date, it will be easy to cross check that a particular certification refers to the current version from the on-chain script addresses and hashes that are also provided.

One thing that might be missing in the CIP description is that our goal is only to be able to certify the on-chain components of a DApp.

Reply:
> it will be easy to cross check that a particular certification refers to the current version from the on-chain script addresses and hashes that are also provided.

IMO this flow should be detailed in this proposal, as to me this is not clear.

Reply:
@RSoulatIOHK if a limitation (disclaimer?) has been placed on the goals of the CIP, I've missed it on the CIP text.  I'm not a dApp expert but what @Ryun1 is saying sounds essential to define the CIP's scope.

Reply:
Resolving this because this sentence (currently in Abstract)
>This CIP descibes a standardised method for certificates to be published and stored on-chain and for stakeholders to be able to verify the different claims of this certificate.

only mentions the on-chain components but nothing about certifying anything else.  As with any other CIP, if it's not specifically mentioned then the CIP doesn't cover it.  @RSoulatIOHK I do believe a limitation in scope should have been clarified explicitly here but I'm not going to hold off on a positive review because of it.